packages/foam-vscode/src/services/variable-resolver.ts

```diff
@@ -199,6 +200,28 @@ export class Resolver {
             )
           );
           break;
+        case 'FOAM_DATE_WEEK':
+          this.promises.set(
+            name,
+            new Promise((resolve) => {
+                // https://en.wikipedia.org/wiki/ISO_8601#Week_dates
+                const date = new Date(this.foamDate);
+
+                // Find Thursday of this week starting on Monday
+                date.setDate(date.getDate() + 4 - (date.getDay() || 7));
+                const thursday = date.getTime();
+
+                // Find January 1st
+                date.setMonth(0); // January
+                date.setDate(1);  // 1st
+                const janFirst = date.getTime();
+
+                // Round the amount of days to compensate for daylight saving time
+                const days = Math.round((thursday - janFirst) / 86400000); // 1 day = 86400000 ms
+                resolve(Math.floor(days / 7) + 1);
```

Comment:
I think this should be in 2 digits format (e.g. `07`) to stay consistent with e.g. the date variable
(we should add a test for this)